edb/server/pgcluster.py

```diff
@@ -767,24 +826,95 @@ def get_remote_pg_cluster(dsn: str) -> RemoteCluster:
 
     loop = asyncio.new_event_loop()
 
-    async def _is_rds():
-        conn = await rcluster.connect()
+    async def _get_cluster_type(
+        conn,
+    ) -> Tuple[Type[RemoteCluster], Optional[str]]:
+        managed_clouds = {
+            'rds_superuser': RemoteCluster,    # Amazon RDS
+        }
+
+        managed_cloud_super = await conn.fetchval(
+            """
+                SELECT
+                    rolname
+                FROM
+                    pg_roles
+                WHERE
+                    rolname = any($1::text[])
+                LIMIT
+                    1
+            """,
+            list(managed_clouds),
+        )
+
+        if managed_cloud_super is not None:
+            return managed_clouds[managed_cloud_super], managed_cloud_super
+        else:
+            return RemoteCluster, None
+
+    async def _detect_capabilities(conn) -> BackendCapabilities:
+        caps = BackendCapabilities.NONE
+
+        try:
+            await conn.execute(f'ALTER SYSTEM SET foo = 10')
+        except asyncpg.InsufficientPrivilegeError:
+            configfile_access = False
+        except asyncpg.UndefinedObjectError:
+            configfile_access = True
+        else:
+            configfile_access = True
+
+        if configfile_access:
+            caps |= BackendCapabilities.CONFIGFILE_ACCESS
+
+        tx = conn.transaction()
+        await tx.start()
+        rname = ''.join(random.choice(string.ascii_letters) for _ in range(10))
```

Comment:
I'd use a UUID here. It's highly unlikely but a collision is possible in which case you'd get an exception you don't currently catch.